File: src/main/java/phoenix/init/PhoenixTiers.java
package phoenix.init;

import net.minecraft.item.IItemTier;
import net.minecraft.item.crafting.Ingredient;

public enum PhoenixTiers implements IItemTier
{
    ZIRCONIUM_TIER(3, 1500, 6.0F, 4.0F, 10, Ingredient.fromItems(PhoenixItems.HIGH_QUALITY_CLAY_ITEM.get()));

    private final int harvestLevel;
    private final int maxUses;
    private final float efficiency;
    private final float attackDamage;
    private final int enchantability;
    private final Ingredient repairMaterial;

    PhoenixTiers(int harvestLevelIn, int maxUsesIn, float efficiencyIn, float attackDamageIn, int enchantabilityIn, Ingredient repairMaterialIn) {
        this.harvestLevel = harvestLevelIn;
        this.maxUses = maxUsesIn;
        this.efficiency = efficiencyIn;
        this.attackDamage = attackDamageIn;
        this.enchantability = enchantabilityIn;
        this.repairMaterial = repairMaterialIn;
    }

    @Override public int getMaxUses()               {  return this.maxUses;        }
    @Override public float getEfficiency()          {  return this.efficiency;     }
    @Override public float getAttackDamage()        {  return this.attackDamage;   }
    @Override public int getHarvestLevel()          {  return this.harvestLevel;   }
    @Override public int getEnchantability()        {  return this.enchantability; }
    @Override public Ingredient getRepairMaterial() { return this.repairMaterial;  }
}

File: src/main/java/phoenix/init/PhoenixTiers.kt
package phoenix.init

import net.minecraft.item.IItemTier
import net.minecraft.item.crafting.Ingredient


enum class PhoenixTiers(private val harvestLevel: Int, private val maxUses: Int, private val efficiency: Float, private val attackDamage: Float, private val enchantability: Int, private val repairMaterial: Ingredient) : IItemTier
{
    ZIRCONIUM_TIER(3, 1500, 6.0f, 4.0f, 10, Ingredient.fromItems(PhoenixItems.HIGH_QUALITY_CLAY_ITEM.get()));

    override fun getEnchantability(): Int = enchantability
    override fun getHarvestLevel  (): Int = harvestLevel
    override fun getMaxUses       (): Int = maxUses
    override fun getEfficiency  (): Float = efficiency
    override fun getAttackDamage(): Float = attackDamage
    override fun getRepairMaterial(): Ingredient = repairMaterial
}